ext/openssl/openssl.c

```diff
@@ -115,6 +117,17 @@ enum php_openssl_cipher_type {
 	PHP_OPENSSL_CIPHER_DEFAULT = PHP_OPENSSL_CIPHER_RC2_40
 };
 
+/* add some encoding rules.  This is normally handled through filters
+ * in the OpenSSL code, but we will do that part as if we were one
+ * of the OpenSSL binaries along the lines of -outform {DER|CMS|PEM}
+ */
+
+enum php_openssl_encoding {
+	ENCODING_DER,
+	ENCODING_CMS,
```

Comment:
shouldn't this be really `ENCODING_SMIME`?

Reply:
Ok.  I called it CMS because of SIME_read_CMS but I think you may be right.  Fixed.